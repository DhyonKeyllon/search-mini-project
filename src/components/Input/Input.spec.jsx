import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { InputComponent } from ".";

describe("<InputComponent />", () => {
  it("should have value of searchValue", () => {
    const fn = jest.fn();

    render(
      <InputComponent
        handleChangeValue={fn}
        searchValue="Testing"
        placeholder="type your search"
      />
    );

    const input = screen.getByPlaceholderText(/type your search/i);

    expect(input).toBeInTheDocument();
    expect(input.value).toBe("Testing");
  });

  it("should call handleChange function on each key pressed", async () => {
    const fn = jest.fn();
    render(
      <InputComponent handleChangeValue={fn} placeholder="type your search" />
    );

    const input = screen.getByPlaceholderText(/type your search/i);
    const valueTyped = "the value";

    await userEvent.type(input, valueTyped);

    expect(input.value).toBe(valueTyped);
    expect(fn).toHaveBeenCalledTimes(valueTyped.length);
  });

  it("should match snapshot", () => {
    const fn = jest.fn();
    const { container } = render(
      <InputComponent
        handleChangeValue={fn}
        searchValue="Testing"
        placeholder="type your search"
      />
    );
    expect(container).toMatchSnapshot();
  });
});
