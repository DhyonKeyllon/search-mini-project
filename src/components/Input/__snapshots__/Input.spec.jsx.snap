// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputComponent /> should match snapshot 1`] = `
<div>
  <input
    class="text-input"
    placeholder="type your search"
    type="search"
    value="Testing"
  />
</div>
`;
